lib/statsd_middleware.rb

```diff
@@ -31,16 +81,35 @@ def call(env)
     if path_parameters
       controller = path_parameters[:controller]
       action = path_parameters[:action]
+      source_app = get_source_app(env)
 
-      duration_tags = ["controller:#{controller}", "action:#{action}"]
-      status_tags = duration_tags + ["status:#{status}"]
-
-      # rubocop:disable Style/RescueModifier
-      StatsD.increment(STATUS_KEY, tags: status_tags) rescue nil
-      StatsD.measure(DURATION_KEY, duration, tags: duration_tags) rescue nil
+      instrument_statsd(status, duration, controller, action, source_app)
     end
 
-    # rubocop:enable Style/RescueModifier
     [status, headers, response]
   end
+
+  private
+
+  def get_source_app(env)
+    source_app = env['HTTP_SOURCE_APP_NAME']
+
+    return nil if source_app.nil?
+    return source_app if SOURCE_APP_NAMES.include?(source_app)
+
+    Raven.capture_message('Unrecognized Source App Request Header',
```

Comment:
Yes it sends to sentry, and yes we want to know when this happens.

You're poking at the fact that a malicious user could load up our sidekiq queues? Since we use sidekiq enterprise, we could enforce some rate limiting / throttling on this call to Sentry.

Alternatively, we could use the Rails.logger and setup a metric log filter + alert.

I think I'll just remove it altogether for now and create a new ticket since knowing when this happens can happen a bit later.